gobotics-frontend/src/App.tsx

```diff
@@ -0,0 +1,124 @@
+import React, {Component} from "react";
+import AssetList from "./components/AssetList";
+import {ShowAddAssetForm, AssetForm} from "./components/AssetForm";
+import Asset from "./types/Asset";
+import Button from '@material-ui/core/Button';
+import { MouseEvent } from 'react';
+
+// todo super(props) ?
+
+
+// hey quick idea for your "showing data" thing that you might find useful
+//as a POF you could have a component that has both the vanilla card + info card stacked
+
+//TODO function components with state how in react
+// React State and props together 
+
+//TODO remove all props:any
+// https://www.typescriptlang.org/docs/handbook/react.html
+// export interface Props {
+// https://stackoverflow.com/questions/52735288/why-does-parameter-props-implicitly-has-an-any-type
+//https://stackoverflow.com/questions/47561848/property-value-does-not-exist-on-type-readonly
+
+
+
+type AppProps = {}
+
+type AppState = { 
+  showAddAssetForm: boolean,
+  addAssetForm: Asset,
+  assets: Array<Asset>
+};
+
+class App extends Component<AppProps, AppState>{
+    constructor(props:AppProps) {
+        super(props)
+        this.state = {
+            'showAddAssetForm': false,
+            'addAssetForm': {
+                'name': '',
+                'location': '',
+                'id' : '',
+                'device_status': 'down'
+            },
+            assets: [
+                {'id': '001', 'name': 'Raspberry Pi', 'device_status': 'up', 'location': 'MIT'},
+                {'id': '002', 'name': 'Linksys Router', 'device_status': 'up', 'location': 'Harvard'},
+                {'id': '003', 'name': 'Linux Laptop', 'device_status': 'down', 'location': 'BU'},
+                {'id' : '004', 'name': 'Arduino', 'device_status': 'up', 'location': 'Binghamton'}
+            ]
+        }
+    }
+    handleAssetFormChange = (event : React.ChangeEvent<HTMLInputElement>) => {
+        const addAssetFormState = this.state.addAssetForm;
+        const { name, value } = event.target;
+        addAssetFormState[name] = value;
+        this.setState({ addAssetForm: addAssetFormState });
+      };
+
+    onSubmit = (event: MouseEvent<HTMLButtonElement>) => {
+        event.preventDefault();
+        console.log(this.state.assets)
+        console.log(this.state.addAssetForm)
+        // TODO ask rohan
+        if (this.state.addAssetForm.name === "" && this.state.addAssetForm.location === "")
```

Comment:
what did you want to ask haha

also, just use double equal instead of triple so that you also check for null values.

Reply:
Addressed in c8cb119